app/src/main/java/com/ecosystem/kin/app/MainActivity.java

```diff
@@ -315,6 +356,9 @@ protected void onDestroy() {
         nativeSpendOrderConfirmationCallback = null;
         nativeEarnOrderConfirmationCallback = null;
         try {
+            if(balanceUpdateObserver != null) {
```

Comment:
do we need it can onDestroy() be called without onStop() called first?

Reply:
yes if somehow the activity got finish onCreate

Reply:
👍